jobmaster/dm/metadata/job.go

```diff
@@ -0,0 +1,99 @@
+package metadata
+
+import (
+	"context"
+
+	"github.com/hanfei1991/microcosm/jobmaster/dm/config"
+	"github.com/hanfei1991/microcosm/lib"
+	"github.com/hanfei1991/microcosm/pkg/adapter"
+	"github.com/hanfei1991/microcosm/pkg/metadata"
+	"github.com/pingcap/errors"
+)
+
+// TODO: use Stage in lib or move Stage to lib.
+type TaskStage int
+
+// These stages may updated in later pr.
+const (
+	StageInit TaskStage = iota
+	StageRunning
+	StagePaused
+	StageFinished
+	StageDelete
+)
+
+// Job represents the state of a job.
+type Job struct {
+	State
```

Comment:
what's the purpose of this embed?

Reply:
Place holder, though now it's interface{}, later we may add some default field or function for `State`